cartographer_ros/cartographer_ros/node.cc

```diff
@@ -535,21 +513,59 @@ cartographer_ros_msgs::StatusResponse Node::FinishTrajectoryUnderLock(
 bool Node::HandleStartTrajectory(
     ::cartographer_ros_msgs::StartTrajectory::Request& request,
     ::cartographer_ros_msgs::StartTrajectory::Response& response) {
-  absl::MutexLock lock(&mutex_);
-  TrajectoryOptions options;
-  if (!FromRosMessage(request.options, &options) ||
-      !ValidateTrajectoryOptions(options)) {
-    const std::string error = "Invalid trajectory options.";
-    LOG(ERROR) << error;
+  const auto trajectory_states = map_builder_bridge_.GetTrajectoryStates();
```

Comment:
Please follow #1262 (update branch) and replace this block up to L527 with sth. like:
```cpp
  // Check if the requested trajectory for the relative initial pose exists.
  response.status = TrajectoryStateToStatus(
      request.relative_to_trajectory_id, {TrajectoryState::ACTIVE, TrajectoryState::FROZEN, TrajectoryState::FINISHED} /* valid states */);
  if (response.status.code != cartographer_ros_msgs::StatusCode::OK) {
    LOG(ERROR) << "Can't start a trajectory with initial pose: " << response.status.message;
    return true;
  }
```

Reply:
Also, this check belongs to the `if (request.use_initial_pose)` block. It shouldn't be executed and throw errors because of default arguments if no initial pose was specified.